### PR TITLE
fix: Properly typecheck webpack-dev-server and fix several undefined issues

### PR DESCRIPTION
**Pull request by agg23**
### User facing changelog
Fixes crashes that could occur when using `webpack` with Component Testing and not passing a custom user config.

### Additional details
Additionally adds proper typechecking to `@cypress/webpack-dev-server`, preventing this type of errors in the future.

### How has the user experience changed?
N/A

### CODE REVIEW

**Review by lmiller1990 — APPROVED** [review]
🎉 

type all the things

File: npm/webpack-dev-server/tsconfig.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
 
     /* Strict Type-Checking Options */
-    "strict": false /* Enable all strict type-checking options. */,
+    "strict": true /* Enable all strict type-checking options. */,
```
Comment (elevatebart) [inline npm/webpack-dev-server/tsconfig.json:27]: isn't typescript type checking strict by default?

Comment (agg23) [reply npm/webpack-dev-server/tsconfig.json:27]: Nope

Screen Shot 2021-05-14 at 10 01 42 AM

Comment (elevatebart) [reply npm/webpack-dev-server/tsconfig.json:27]: This is odd indeed...
Thank you for checking

**Review by elevatebart — APPROVED** [review]

File: npm/webpack-dev-server/tsconfig.json (hunk 1 above)
Comment (lmiller1990) [reply npm/webpack-dev-server/tsconfig.json:27]: If you create a new `tsconfig.json` with `yarn tsc --init` is sets `"strict": true`.

**Comment by jennifer-shehane** [comment]
This looks like it's just affecting an npm package and not part of the binary release. Untagging for 7.4.0 release.

**Pull request merged**